### PR TITLE
RHCLOUD-22557 Using Failed instead of FailedInternal and FailedExternal

### PR DESCRIPTION
**Pull request by josejulio**

### CODE REVIEW

**Comment by josejulio** [comment]
Marking as draft - I'm missing the inverse operation for a filter

File: backend/src/main/java/com/redhat/cloud/notifications/routers/EventResource.java [hunk 1]
```diff
@@ -64,6 +66,7 @@ public Page<EventLogEntry> getEvents(@Context SecurityContext securityContext, @
 
         Set<EndpointType> basicTypes = Collections.emptySet();
         Set<CompositeEndpointType> compositeTypes = Collections.emptySet();
+        Set<NotificationStatus> notificationStatusSet = status == null ? Set.of() : toNotificationStatus(status);
```
Comment (josejulio) [inline backend/src/main/java/com/redhat/cloud/notifications/routers/EventResource.java:69]: I noticed that `status` was always an empty set, even if the query param wasn't specified- Anyway added a null check.

File: backend/src/main/java/com/redhat/cloud/notifications/routers/EventResource.java [hunk 2]
```diff
@@ -130,4 +133,48 @@ public Page<EventLogEntry> getEvents(@Context SecurityContext securityContext, @
...
+            default:
+                Log.warnf("Uncovered status found:[%s]. This is a bug", status);
+                return EventLogEntryActionStatus.UNKNOWN;
+        }
+    }
+
+    static Set<NotificationStatus> toNotificationStatus(Set<EventLogEntryActionStatus> statusSet) {
+        Set<NotificationStatus> notificationStatusSet = new HashSet<>();
+        statusSet.forEach(status -> {
+            switch (status) {
```
Comment (gwenneg) [inline backend/src/main/java/com/redhat/cloud/notifications/routers/EventResource.java:157]: Could you please add a null-check on `status` here? RESTEasy allows `null` values in collections that are received as REST params. We'll probably need to throw a `BadRequestException` is status is `null`.

File: backend/src/main/java/com/redhat/cloud/notifications/routers/EventResource.java [hunk 3]
```diff
@@ -130,4 +133,48 @@ public Page<EventLogEntry> getEvents(@Context SecurityContext securityContext, @
...
+            case FAILED_EXTERNAL:
+            case FAILED_INTERNAL:
+                return EventLogEntryActionStatus.FAILED;
+            default:
+                Log.warnf("Uncovered status found:[%s]. This is a bug", status);
+                return EventLogEntryActionStatus.UNKNOWN;
+        }
+    }
+
+    static Set<NotificationStatus> toNotificationStatus(Set<EventLogEntryActionStatus> statusSet) {
```
Comment (gwenneg) [inline backend/src/main/java/com/redhat/cloud/notifications/routers/EventResource.java:154]: We could use a test for each switch case.

File: backend/src/main/java/com/redhat/cloud/notifications/routers/EventResource.java (hunk 2 above)
Comment (josejulio) [reply backend/src/main/java/com/redhat/cloud/notifications/routers/EventResource.java:162]: done

File: backend/src/main/java/com/redhat/cloud/notifications/routers/EventResource.java (hunk 3 above)
Comment (josejulio) [reply backend/src/main/java/com/redhat/cloud/notifications/routers/EventResource.java:155]: aaand done.

**Pull request merged**